### ISSUE TITLE
100% Crash at Boot when FPP Device entered by Hostname is not online [Mac]

### ISSUE THREAD

**Issue by kylegrymonprez** [xLightsSequencer/xLights#4968]
**Describe the bug**
If a FPP device is added by hostname and isn't reachable, then xLights crashes

**To Reproduce**
1. Add a new Ethernet Controller
2. Set Vendor to FPP
3. Set Variant to FPP Player Only
4. Enter a hostname in the IP Address field (I used "fpp-planter.local")
5. Reboot xLights
6. xLights crashes

**Expected behavior**
xLights should be able to handle an unreachable FPP device

**Versions (please complete the following information):**
 - OS: MacOS 15.1
 - xLights version TestFlight, but was happening since 2024.18, maybe 2024.17

**Additional context**

**Attachments**
Logs: [xLights_dbgrpt-75688-20241118T130913.zip](https://github.com/user-attachments/files/17806315/xLights_dbgrpt-75688-20241118T130913.zip)

**Comment by cybercop23** [comment]
Hmmm... Don't you have a Unifi network too? Both Dan and I have.. Here's my prod xlights and what's actually on on this hour from FPP... No crashy crash...
![image](https://github.com/user-attachments/assets/6b64f217-3890-4745-882a-ffde59a48f07)

**Comment by kylegrymonprez** [comment]
Yeah I have unifi too - but if I add an FPP device via hostname and it can't resolve, it crashes shortly after launch. Once I delete the controller manually in XML, it stops crashing. Seems semi connected to the other hostname IP issues going around.

**Comment by cybercop23** [comment]
> Yeah I have unifi too - but if I add an FPP device via hostname and it can't resolve, it crashes shortly after launch. Once I delete the controller manually in XML, it stops crashing. Seems semi connected to the other hostname IP issues going around.

I think that's what I have... I have all these contrllers that aren't on, but I jsuts started xLights and no crashy crash. Only the ones on the right in FPP are currently connected.

**Comment by kylegrymonprez** [comment]
`2024-11-18 13:09:13,248 6354579456 log_base [ERROR] Failure to access http://fpp-planter.local/: 28: Timeout was reached.
2024-11-18 13:09:13,255 6355726336 log_base [CRIT] Crashed: Fatal exception occurred.
2024-11-18 13:09:14,796 6355726336 log_base [CRIT] xLights version 2024.18.1
Time: 2024-11-18T13:09:14
<email>wacamoe@gmail.com</email>
Crashed Thread ID: 0x17ad4b000

0   xLights                             0x0000000101e2625c _ZN14xlCrashHandler11HandleCrashEbRKNSt3__112basic_stringIcNS0_11char_traitsIcEENS0_9allocatorIcEEEE + 3672
1   xLights                             0x0000000102a701cc _ZN9xlBaseApp16OnFatalExceptionEv + 88
2   xLights                             0x0000000102c8e660 _ZL20wxFatalSignalHandleri + 36
3   libsystem_platform.dylib            0x000000019b1c0184 _sigtramp + 56
4   libc++.1.dylib                      0x000000019b0c2758 _ZNSt3__15mutex4lockEv + 16
5   libc++.1.dylib                      0x000000019b0c2758 _ZNSt3__15mutex4lockEv + 16
6   libc++.1.dylib                      0x000000019b0c2ec0 _ZNSt3__119__shared_mutex_base4lockEv + 40
7   xLights                             0x00000001027c5d24 _ZN6Output13SetResolvedIPERKNSt3__112basic_stringIcNS0_11char_traitsIcEENS0_9allocatorIcEEEE + 60
8   xLights                             0x0000000102b0dba0 _ZN8ip_utils10ResolveJob7ProcessEv + 608
9   xLights                             0x00000001029d04d8 _ZL9startFuncP13JobPoolWorker + 940
10  xLights                             0x00000001029d24d4 _ZNSt3__114__thread_proxyB8ne180100INS_5tupleIJNS_10unique_ptrINS_15__thread_structENS_14default_deleteIS3_EEEEPFvP13JobPoolWorkerES8_EEEEEPvSC_ + 52
11  libsystem_pthread.dylib             0x000000019b18b2e4 _pthread_start + 136
12  libsystem_pthread.dylib             0x000000019b1860fc thread_start + 8`

Relevant Crash

**Comment by derwin12** [comment]
Yea this crash looks to be in the same code that crashes my dev env.

**Comment by kylegrymonprez** [comment]
I just tested and i'm not crashing anymore in prod or dev - dunno if it's still an issue for others.

**Comment by cybercop23** [comment]
I think @derwin12 also said its no longet crashing. Please fell free to reopen if it comes back.
